File: app/src/main/java/ch/bailu/aat/gpx/linked_list/List.java
package ch.bailu.aat.gpx.linked_list;

public class List {
    private Node first=null;
    private Node last=null;
    
    private int count=0;
    
    public void append(Node node) {
        node.setPrevious(last);
        
        if (first==null) {
            first=node;
        } else { 
            last.setNext(node);
        }
        
        last=node;
        count++;
    }
    
    
    public Node getFirst() {
        return first;
    }
    
    
    public Node getLast() {
        return last;
    }
    
    
    public int size() {
        return count;
    }
    
    
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/gpx/linked_list/List.kt
package ch.bailu.aat_lib.gpx.linked_list

class List {
    var first: Node? = null
        private set

    var last: Node? = null
        private set

    private var count = 0

    fun append(node: Node) {
        node.previous = last
        if (first == null) {
            first = node
        } else {
            last!!.next = node
        }
        last = node
        count++
    }

    fun size(): Int {
        return count
    }
}
